src/operations/distinct.ts

```diff
@@ -96,7 +96,7 @@ export class DistinctOperation extends CommandOperation<any[]> {
 
     const result = await super.executeCommand(server, session, cmd, timeoutContext);
 
-    return this.explain ? result : result.values;
+    return this.explain ? (result as any[]) : result.values;
```

Comment:
```suggestion
    // @ts-expect-error Explain always returns a document
    return this.explain ? result : result.values;
```

Instead of hiding the casting away the here, we should make it clear we expect this TS error and explain why

Reply:
Updated.